### ISSUE TITLE
Topology translation test case fails randomly due to assumed ordering

### ISSUE THREAD

**Issue by jiahuif** [kubernetes/kubernetes#87163]
**What happened**:
Received the following test failure during unit test execution
```
--- FAIL: TestTopologyTranslation (0.00s)
    translate_test.go:171: Testing GCE PD with zone labels
    translate_test.go:171: Testing GCE PD with existing topology (beta keys)
    translate_test.go:171: Testing GCE PD with existing topology (CSI keys)
    translate_test.go:171: Testing GCE PD with zone labels and topology
    translate_test.go:171: Testing GCE PD with regional zones
    translate_test.go:181: Expected node affinity {&NodeSelector{NodeSelectorTerms:[]NodeSelectorTerm{NodeSelectorTerm{MatchExpressions:[]NodeSelectorRequirement{NodeSelectorRequirement{Key:topology.gke.io/zone,Operator:In,Values:[europe-west1-b europe-west1-c],},},MatchFields:[]NodeSelectorRequirement{},},},}}, got {&NodeSelector{NodeSelectorTerms:[]NodeSelectorTerm{NodeSelectorTerm{MatchExpressions:[]NodeSelectorRequirement{NodeSelectorRequirement{Key:topology.gke.io/zone,Operator:In,Values:[europe-west1-c europe-west1-b],},},MatchFields:[]NodeSelectorRequirement{},},},}}
    translate_test.go:192: Expected node affinity {&NodeSelector{NodeSelectorTerms:[]NodeSelectorTerm{NodeSelectorTerm{MatchExpressions:[]NodeSelectorRequirement{NodeSelectorRequirement{Key:topology.gke.io/zone,Operator:In,Values:[europe-west1-b europe-west1-c],},},MatchFields:[]NodeSelectorRequirement{},},},}}, got {&NodeSelector{NodeSelectorTerms:[]NodeSelectorTerm{NodeSelectorTerm{MatchExpressions:[]NodeSelectorRequirement{NodeSelectorRequirement{Key:topology.gke.io/zone,Operator:In,Values:[europe-west1-c europe-west1-b],},},MatchFields:[]NodeSelectorRequirement{},},},}}
    translate_test.go:171: Testing GCE PD with regional topology
    translate_test.go:171: Testing GCE PD with regional zone and topology
    translate_test.go:171: Testing GCE PD with multiple node selector terms
    translate_test.go:171: Testing AWS EBS with zone labels
    translate_test.go:171: Testing AWS EBS with zone labels and topology
    translate_test.go:171: Testing OpenStack Cinder with zone labels
    translate_test.go:171: Testing OpenStack Cinder with zone labels and topology
FAIL
FAIL	k8s.io/kubernetes/vendor/k8s.io/csi-translation-lib	0.071s

```
**What you expected to happen**:
aforementioned test case passes consistently. 
**How to reproduce it (as minimally and precisely as possible)**:
Run the affected test case multiple times locally. I am still reporting this as a bug because it involves functionality of implementation code.
**Anything else we need to know?**:

root cause is as follows
https://github.com/kubernetes/kubernetes/blob/3a7e0c252b6b363abe5fc2fbaa92047401d694ce/staging/src/k8s.io/csi-translation-lib/plugins/in_tree_volume.go#L111-L144

at line 122
```go
zones = filteredZones.UnsortedList()
```
This will cause `zones` to have undefined ordering. However, in the test, the ordering is assumed when creating the expected value, being "europe-west1-b" before "europe-west1-c"
https://github.com/kubernetes/kubernetes/blob/df7a3f976afa82191d6b90be4d01864a2a061a35/staging/src/k8s.io/csi-translation-lib/translate_test.go#L121-L124

This case will pass if the translated value has the assumed ordering.

I would like to change 
```go
zones = filteredZones.UnsortedList()
```
to
```go
zones = filteredZones.List()
```

Let me know if I can proceed with such change. Or, if the original behavior is required for some reason, we can change the test instead.

/kind bug
/sig storage

**Comment by chymy** [comment]
I encountered the same problem.

**Comment by jiahuif** [comment]
Fixed by #88987 (cherry-picked to 1.18 #89238)
/close
